common/include/common/common_utils.hpp

```diff
@@ -21,7 +21,12 @@
 #include <common/float_comparison.hpp>
 #include <common/system_reference_blas.hpp>
 
+#ifdef BLAS_BENCHMARK_INDEX_T
```

Comment:
You can always define this on the Cmake and avoid having it in the code. Set the default value for BLAS_BENCHMARK_INDEX_T to be int in CMAKE and directly set `using index_t=BLAS_BENCHMARK_INDEX_T`

Reply:
In [107e289](https://github.com/codeplaysoftware/sycl-blas/pull/339/commits/107e289ed09eef9c570e5cda6fd6fe9f909139cc):
* The default values for `BLAS_BENCHMARK_INDEX_T` have been removed from the .hpp files.
  * The CMake should always define something - there should be no need for a default in the CMake.
* For the unit tests, the `BLAS_BENCHMARK_INDEX_T` has been renamed to `BLAS_TEST_INDEX_T`.

Reply:
Is there any reason that we need to use two different one? can we call it BLAS_INDEX_T and use it for all?

Reply:
Possibly. The intention was to avoid confusion with `BLAS_INDEX_TYPES`, but instead they may have confused things.
* `BLAS_INDEX_TYPES` can support multiple types eg. `-DBLAS_INDEX_TYPES="int32_t;int64_t"`.
* The unit tests and benchmarks are not written to support testing with multiple index types.
  * This would require far more substantial modifications than this PR.
  * Instead, the first type of the list of `BLAS_INDEX_TYPES` is used.

In [3c27a08](https://github.com/codeplaysoftware/sycl-blas/pull/339/commits/3c27a08309f7f507aac7f8a31b3836b0f3edd8d0) I've renamed them as `BLAS_INDEX_T` as suggested.

Reply:
Yes, I did not meant to touch BLAS_INDEX_TYPES. I just wanted you to use BLAS_INDEX_T instead of BLAS_BENCHMARK_INDEX_T and BLAS_TEST_INDEX_T.